### PR TITLE
Remove IQ# install from notebook validation script

### PR DESCRIPTION
**Pull request by tcNickolas**
This script is used in two scenarios: on a CI build agent, the build jobs run install-iqsharp.ps1 first, so repeating IQ# install is not necessary, and when running locally, the user is likely to have IQ# installed by other means, so reinstalling it causes version conflicts down the road.

### CODE REVIEW

**Comment by tcNickolas** [comment]
@anpaz Could you please take a look at this PR? Your main concern was that the internal E2E build relies on this script installing IQ#, so I ran it for this change [here](https://dev.azure.com/ms-quantum-public/Microsoft%20Quantum%20(public)/_build/results?buildId=25914&view=results) to make sure it passes. And it passes the Katas CI and does not install IQ# locally, so it does what I want.

File: scripts/validate-notebooks.ps1 [hunk 1]
```diff
@@ -34,7 +34,14 @@ Param(
 )
 
 
-& "$PSScriptRoot/install-iqsharp.ps1"
+# Detect if we're running on Azure Pipelines.
+$IsAzurePipelines = "$Env:TF_BUILD" -ne "";
```
Comment (anpaz) [inline scripts/validate-notebooks.ps1:38]: We normally use the AGENT_ID variable for this.

**Review by anpaz — APPROVED** [review]
LGTM. The only minor change is to use the AGENT_ID variable (instead of TF_BUILD) since that's what we use in other places and I'd like to keep it consistent.

File: scripts/validate-notebooks.ps1 (hunk 1 above)
Comment (tcNickolas) [reply scripts/validate-notebooks.ps1:38]: Sure, I can use that one instead

**Pull request merged**